### ISSUE TITLE
Publishing as NativeAOT should not depend on AppHost

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#79575]
We currently need to build apphost because for some reason PublishAot ends up needing it in some of our test legs:

https://github.com/dotnet/runtime/blob/cdb9f5f179db22745fbb2e878460662c0b21643e/eng/pipelines/runtime.yml#L206

Without this line, the CI fails with:

```
Could not copy the file "/__w/1/s/artifacts/bin/trimmingTests/projects/System.Diagnostics.DiagnosticSource.NativeAotTests/DiagnosticSourceEventSourceTests/linux-x64/obj/Release/net8.0/linux-x64/apphost" because it was not found.
```

This is probably not just a test issue, but also a product issue. Historically the NativeAOT targets were just an addon to the SDK targets that hacked into the SDK (we e.g. delete outputs the SDK produces in publish and replace them with NativeAOTd stuff).

The SDK probably needs to become more aware of what NativeAOT does and suppress some behaviors.

**Comment by ViktorHofer** [comment]
That error can be avoided by setting `<UseAppHost>false</UseAppHost>` but that will then raise the next error:

```
C:\Program Files\dotnet\sdk\7.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.RuntimeIdentifierInference.targets(186
,5): error NETSDK1067: Self-contained applications are required to use the application host. Either set SelfContained t
o false or set UseAppHost to true. [C:\temp\useaphost\useaphost.csproj]
```

**Comment by MichalStrehovsky** [comment]
Yeah, the SDK needs to be taught that PublishAot is sort of like self contained, but it doesn't need an apphost. I probably should have opened this in the SDK repo because the main fix will likely be in the SDK, but SDK repo is where issues typically go die of neglect and loneliness.

**Comment by ViktorHofer** [comment]
If we are looking for an easy solution, then this would just require to modifications in the SDK:
-  Add `'$(PublishAot)' != 'true'` to this condition: https://github.com/dotnet/sdk/blob/main/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L200-L201
- Make `UseAppHost` default to false when `'$(PublishAot)' == 'true'`

**Comment by jkotas** [comment]
> Make UseAppHost default to false when '$(PublishAot)' == 'true'

We do want to keep using app host for `dotnet run` or `F5` in Visual Studio. I do not think we want to change the global default to false when PublishAot is set.
